### PR TITLE
Avoid too much retries if login failed

### PR DESCRIPTION
**Pull request by j0k3r**
If user provide bad login information, there will be too much attempt to login because login will always fail.
Instead, just try 2 times and skip if we can't login.

I tried to add some spec but failed.
I don't know how to test the multiple calls. I tried with `shouldBeCalledTimes` without success.

### CODE REVIEW

File: lib/Guzzle/AuthenticatorSubscriber.php [hunk 1]
```diff
@@ -83,6 +83,8 @@ public function loginIfRequested(CompleteEvent $event)
             $emitter->attach($this);
 
             $event->retry();
+
+            ++self::$retries;
```
Comment (bdunogier) [inline lib/Guzzle/AuthenticatorSubscriber.php]: I'd replace the condition above with: 

´´´php
if (++self::$retries >= self::MAX_TRIES) {
  break;
}
´´´

Comment (j0k3r) [inline lib/Guzzle/AuthenticatorSubscriber.php]: I can't use `break` in that case :thinking:

Comment (bdunogier) [inline lib/Guzzle/AuthenticatorSubscriber.php]: Review on mobile... i'll have a 2nd look on a proper screen ^^

Comment (bdunogier) [inline lib/Guzzle/AuthenticatorSubscriber.php]: Indeed, nevermind.

**Pull request merged**